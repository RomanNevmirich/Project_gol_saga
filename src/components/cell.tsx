import styled from 'styled-components';
import React from "react";

const StyledCell = styled.div<{value: number}>`
    display: block;
    width: 12px;
    height: 12px;
    border: solid;
    border-width: 1px;
    border-color: black;
    will-change: background-color, border-color;
    transition: border-color 0.1s;
    background-color: ${(props) => (props.value ? 'cadetblue' : 'white')};
`;

type cell = {
    value: 0|1
}

export const Cell = ( value: cell ) => (
    <StyledCell value={value.value} ></StyledCell>
);

export default Cell;