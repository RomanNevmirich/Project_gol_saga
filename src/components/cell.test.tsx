import React from 'react';
import { render, screen, fireEvent, createEvent } from '@testing-library/react';
import { Cell } from './cell';

describe('Cell component check', () => {
  
  it('Cell render fill', () => {
    const { asFragment } = render(<Cell value={1}></Cell>);

    expect(asFragment()).toMatchSnapshot();
  });
  it('Cell render empty', () => {
    const { asFragment } = render(<Cell value={0}></Cell>);

    expect(asFragment()).toMatchSnapshot();
  });
});