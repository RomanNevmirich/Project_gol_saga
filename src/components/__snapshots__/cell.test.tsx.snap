// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component check Cell render empty 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ hcpULb"
    value="0"
  />
</DocumentFragment>
`;

exports[`Cell component check Cell render fill 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ klZATh"
    value="1"
  />
</DocumentFragment>
`;